src/pages/Refund.tsx

```diff
@@ -29,19 +30,32 @@ const Refund = () => {
 
         const allSwaps = await getSwaps();
 
-        const swapsToRefund = allSwaps
-            .filter(refundSwapsSanityFilter)
-            .filter(
-                (swap) =>
-                    swapStatusFailed.TransactionLockupFailed === swap.status,
-            );
+        const swapsWithUTXO = (
+            await Promise.all(
+                allSwaps.filter(refundSwapsSanityFilter).map(async (swap) => {
```

Comment:
Same thing here. You cannot hit the block explorer for every swap. It will get you rate limited